base/setup/usetup/usetup.c

```diff
@@ -626,6 +626,174 @@ UpdateKBLayout(VOID)
     }
 }
 
+/*
+ * Function Name: ProgressTimeOutStringHandler
+ *
+ * Purpose: Handles the generation (displaying) of the timeout countdown to the screen
+ *          dinamically
+ *
+ * Return: FALSE
+ */
+static
+BOOLEAN NTAPI
+ProgressTimeOutStringHandler(
+    IN PPROGRESSBAR Bar,
+    IN BOOLEAN AlwaysUpdate,
+    OUT PSTR Buffer,
```

Comment:
RtlStringCchWhatever always end the buffer with a NULL-terminator, so the out string will always be null-terminated, so PSTR is good here.

Reply:
It won't be null-terminated on entry to the function, therefore it's not always null-terminated.

Reply:
Well here we don't care what the buffer is in entry, but what it is in output. Exactly like the RtlStringCch/CbWhatever functions that say _Out_ LPWSTR lpBuffer.

Reply:
And those functions specifically have `_Always_(_Post_z_)` annotations.
Really, what the _Null_terminated_ annotation does is it makes Windbg display the variable as a string. Which is something you _don't_ want for an output parameter because it's not initialized on input and would show garbage.